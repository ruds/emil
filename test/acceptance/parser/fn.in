fn x => x;
fn (0, bar) => bar
 | (1, bar) => (-) (bar, 1)
 | (_, bar) => Math.pow bar 2;
