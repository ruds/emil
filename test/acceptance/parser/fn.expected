0001 (ExprTopDecl
         :expr
         (FnExpr
             :cases
             (((IdentifierPattern
                 :qualifiers
                 ()
                 :identifier
                 "78"
                 :is_op
                 false
                 :is_prefix_op
                 false)
               . (IdentifierExpr
                     :qualifiers
                     ()
                     :identifier
                     "78"
                     :is_op
                     false
                     :is_prefix_op
                     false)))))
0002 (ExprTopDecl
         :expr
         (FnExpr
             :cases
             (
                 ((TuplePattern
                     :patterns
                     (
                         (LiteralPattern
                             :val
                             (BigintLiteralExpr
                                 :val
                                 0))
                         (IdentifierPattern
                             :qualifiers
                             ()
                             :identifier
                             "626172"
                             :is_op
                             false
                             :is_prefix_op
                             false)
                     ))
                   . (IdentifierExpr
                         :qualifiers
                         ()
                         :identifier
                         "626172"
                         :is_op
                         false
                         :is_prefix_op
                         false))
                 ((TuplePattern
                     :patterns
                     (
                         (LiteralPattern
                             :val
                             (BigintLiteralExpr
                                 :val
                                 1))
                         (IdentifierPattern
                             :qualifiers
                             ()
                             :identifier
                             "626172"
                             :is_op
                             false
                             :is_prefix_op
                             false)
                     ))
                   . (ApplicationExpr
                         :exprs
                         (
                             (IdentifierExpr
                                 :qualifiers
                                 ()
                                 :identifier
                                 "2D"
                                 :is_op
                                 true
                                 :is_prefix_op
                                 false)
                             (TupleExpr
                                 :exprs
                                 (
                                     (IdentifierExpr
                                         :qualifiers
                                         ()
                                         :identifier
                                         "626172"
                                         :is_op
                                         false
                                         :is_prefix_op
                                         false)
                                     (BigintLiteralExpr
                                         :val
                                         1)
                                 ))
                         )))
                 ((TuplePattern
                     :patterns
                     (
                         (WildcardPattern)
                         (IdentifierPattern
                             :qualifiers
                             ()
                             :identifier
                             "626172"
                             :is_op
                             false
                             :is_prefix_op
                             false)
                     ))
                   . (ApplicationExpr
                         :exprs
                         (
                             (IdentifierExpr
                                 :qualifiers
                                 ("4D617468")
                                 :identifier
                                 "706F77"
                                 :is_op
                                 false
                                 :is_prefix_op
                                 false)
                             (IdentifierExpr
                                 :qualifiers
                                 ()
                                 :identifier
                                 "626172"
                                 :is_op
                                 false
                                 :is_prefix_op
                                 false)
                             (BigintLiteralExpr
                                 :val
                                 2)
                         )))
             )))
0005 (EndOfFileTopDecl)
